mmv1/templates/terraform/pre_create/dialogflowcx_set_location_skip_default_obj.go.erb

```diff
@@ -0,0 +1,42 @@
+
+// extract location from the parent
+location := ""
+
+if parts := regexp.MustCompile(`locations\/([^\/]*)\/`).FindStringSubmatch(d.Get("parent").(string)); parts != nil {
+    location = parts[1]
+} else {
+    return fmt.Errorf(
+        "Saw %s when the parent is expected to contains location %s",
+        d.Get("parent"),
+        "projects/{{project}}/locations/{{location}}/...",
+    )
+}
+
+url = strings.Replace(url,"-dialogflow",fmt.Sprintf("%s-dialogflow",location),1)
+
+// if it's a default object Dialogflow creates for you, "Update" instead of "Create"
+isDefaultStartFlow, _ := d.Get("is_default_start_flow").(bool)
+isDefaultWelcomeIntent, _ := d.Get("is_default_welcome_intent").(bool)
+isDefaultNegativeIntent, _ := d.Get("is_default_negative_intent").(bool)
```

Comment:
to be extremely pedantic, "the ignored `_` is an error that will be not-nil and the boolean will have the default value of false, so it all works out"

Reply:
I'm glad you highlighted that, I glossed over the fact that those `_` are preventing the provider experiencing a panic here.

I think add whatever comment you think makes sense, so that maintainers know what to be aware of

Reply:
I added that we're deliberately ignoring the error in the missing field case; I also added these words to the pre-delete code, which I realized I'd missed before. Thanks!